### PR TITLE
enabled fp16 tests that work with compiler fixes in test_nn

### PR DESCRIPTION
**Pull request by lcskrishna**

### CODE REVIEW

File: test/test_nn.py [hunk 1]
```diff
@@ -4918,6 +4918,7 @@ def test_pdist_empty_row(self):
             inp = torch.randn(1, 3, dtype=torch.double, device=device, requires_grad=True)
             self.assertTrue(gradcheck(F.pdist, (inp,)))
 
+    @skipIfRocm
```
Comment (iotamudelta) [inline test/test_nn.py]: Why are we now skipping this?

Comment (lcskrishna) [reply test/test_nn.py]: it currently fails with the latest compiler changes.

Comment (iotamudelta) [reply test/test_nn.py]: Interesting, I hadn't seen that. Let's see if the CI also confirms this once it's again up...

**Comment by jithunnair-amd** [comment]
I couldn't figure out why the rocm build failed because log is too long due to rocrand warnings, so can't view full log properly in browser. Truncated version didn't contain the error reason. Any other tricks?

**Comment by iotamudelta** [comment]
It failed b/c the entire CI is currently busted. You can get the log as text, that'll work. But yes, indeed, we need to do something about the verbosity.

**Comment by iotamudelta** [comment]
Ignoring well-known cuda8 cudnn6 failure - merging

**Pull request merged**